glusterd2/xlator/options/options.go

```diff
@@ -236,7 +242,17 @@ func ValidatePriorityOrSize(o *Option, val string) error {
 
 // ValidateSizeList validates if the option is a valid size list
 func ValidateSizeList(o *Option, val string) error {
-	return ErrInvalidArg
+	l := len(val)
+	if strings.ContainsRune(val, 'B') {
+		t := strings.TrimSpace(val[:l-2])
+		v, err := strconv.ParseInt(t, 10, 64)
+		if err != nil {
+			return err
+		} else if v%512 != 0 {
```

Comment:
The blocksize validation being done here, should be done on the parsed final size in bytes. Not just the digits left after removing the `*B` suffix. There are packages which do human-friendly size parsing which you can use to do this, or you could write your own parse routine that matches `gf_string2bytesize` (more specifically `gf_string2bytesize_range`).